### PR TITLE
drivers: regulator: npm6001: Fix compiler warning

### PR DESCRIPTION
**Pull request by balaji-srin**
This patch fixes a compiler warning about the val being used uninitialized. The previously present if check did not seem to satisfy compiler. Hence the val is now explictly initialized.

### CODE REVIEW

**Review by maxd-nordic — APPROVED** [review]

File: drivers/regulator/regulator_npm6001.c [hunk 1]
```diff
@@ -221,7 +221,7 @@ static int regulator_npm6001_ldo0_set_voltage(const struct device *dev, int32_t
 		}
 	}
 
-	if (i == ARRAY_SIZE(ldo0_voltages)) {
+	if (val == 0xff) {
```
Comment (gmarull) [inline drivers/regulator/regulator_npm6001.c:224]: why is this needed?

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/regulator/regulator_npm6001.c (hunk 1 above)
Comment (balaji-nordic) [reply drivers/regulator/regulator_npm6001.c:224]: I have tried to explain in the commit message and hence the PR description

Comment (gmarull) [reply drivers/regulator/regulator_npm6001.c:224]: Sorry, but how is this change related to uninitialized variable? This is changing exit logic, which should be fine as is.

Comment (balaji-nordic) [reply drivers/regulator/regulator_npm6001.c:224]: `val == 0fxff` is more explicit than `i == ARRAY_SIZE(ldo0_votages)`  and hence the compiler does not complain. Without this change, the compiler throws the following error.

/home/balaji/oob-firmware/zephyr/drivers/regulator/regulator_npm6001.c:228:16: warning: 'val' may be used uninitialized [-Wmaybe-uninitialized]
  228 |         return i2c_reg_write_byte_dt(&config->i2c, NPM6001_LDO0VOUT, val);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/balaji/oob-firmware/zephyr/drivers/regulator/regulator_npm6001.c:214:17: note: 'val' was declared here
  214 |         uint8_t val;

Comment (aasinclair) [reply drivers/regulator/regulator_npm6001.c:224]: This error doesn't relate to val == 0xff, just the definition of val.
Do you get a warning if you just initialise val to zero?

Comment (balaji-nordic) [reply drivers/regulator/regulator_npm6001.c:224]: I don't get the warning if I initialize it to 0. It just needs some init value. I believe that 0xff is the correct (invalid) init value for this variable.

Comment (gmarull) [reply drivers/regulator/regulator_npm6001.c:224]: if initialized to zero (instead of 0xff, which initially looks like a magic value) will resolve the issue.

**Review by simensrostad — APPROVED** [review]

File: drivers/regulator/regulator_npm6001.c (hunk 1 above)
Comment (aasinclair) [reply drivers/regulator/regulator_npm6001.c:224]: If you initialise val to zero, and leave line 224 unchanged, do you still get the warning?

Comment (balaji-nordic) [reply drivers/regulator/regulator_npm6001.c]: No. I have now pushed the change where I only initialize the `val` to `0`. Left line 224 unchanged.

**Review by aasinclair — DISMISSED** [review]

File: drivers/regulator/regulator_npm6001.c [hunk 2]
```diff
@@ -211,7 +211,7 @@ static int regulator_npm6001_ldo0_set_voltage(const struct device *dev, int32_t
 					      int32_t max_uv)
 {
 	const struct regulator_npm6001_config *config = dev->config;
-	uint8_t val;
+	uint8_t val = 0;
```
Comment (gmarull) [inline drivers/regulator/regulator_npm6001.c:214]: nit 
```suggestion
	uint8_t val = 0U;
```

**Review by gmarull — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]
